project/dexter-peer-review/src/java/com/samsung/sec/dexter/peerreview/plugin/DexterPeerReviewPlugin.java

```diff
@@ -0,0 +1,311 @@
+		int[] offsets = new int[sourcecode.length()];
+
+		int line = 1;
+		Pattern pattern = Pattern.compile("\n");
+		Matcher matcher = pattern.matcher(sourcecode);
+		matcher.region(0, sourcecode.length());
+		while (matcher.find()) {
+			offsets[line + 1] = matcher.end();
+			line++;
+		}
+		int[] newOffsets = createNewArray(offsets, line);
+		return newOffsets;
+	}
+
+	@Override
+	public AnalysisResult analyze(AnalysisConfig config) {
+		final CharSequence sourcecode = config.getSourcecodeThatReadIfNotExist();
+		AnalysisResult result = analysisEntityFactory.createAnalysisResult(config);
+
+		try {
+			int[] offsets = makeOffsetArray(sourcecode);
+			List<DPRComment> comments = getAllDPRCommentFromSourcecode(offsets, sourcecode);
+			List<Defect> defectList = makeDefectList(config, comments);
+			result.setDefectList(defectList);
+		} catch (Exception e) {
+			logger.error("incorrect regExp:");
+			logger.error(e.getMessage(), e);
+
+		}
+
+		return result;
+	}
+
+	private List<Defect> makeDefectList(AnalysisConfig config, List<DPRComment> comments) {
+		ArrayList<DPRComment> criDefects = Lists.newArrayList();
+		ArrayList<DPRComment> majDefects = Lists.newArrayList();
+		ArrayList<DPRComment> crcDefects = Lists.newArrayList();
+
+		for (DPRComment comment : comments) {
+			if ("CRI".equals(comment.getSeverity())) {
+				criDefects.add(comment);
+			} else if ("MAJ".equals(comment.getSeverity())) {
+				majDefects.add(comment);
+			} else {
+				crcDefects.add(comment);
+			}
+		}
+
+		List<Defect> defectList = new ArrayList<Defect>();
+		if (!criDefects.isEmpty()) {
+			defectList.add(makeDefect(config, "CRI", criDefects));
+		}
+		if (!majDefects.isEmpty()) {
+			defectList.add(makeDefect(config, "MAJ", majDefects));
+		}
+		if (!crcDefects.isEmpty()) {
+			defectList.add(makeDefect(config, "CRC", crcDefects));
+		}
+
+		return defectList;
+	}
+
+	private Defect makeDefect(AnalysisConfig config, String severity, ArrayList<DPRComment> defectInfos) {
+		Defect defect = new Defect();
+
+		defect.setCheckerCode("DPR_" + severity);
+		defect.setFileName(config.getFileName());
+		defect.setModulePath(config.getModulePath());
+		defect.setClassName("");
+		defect.setMethodName("");
+		defect.setLanguage(config.getLanguageEnum().toString());
+		defect.setSeverityCode(severity);
+		defect.setCategoryName("PeerReview");
+		defect.setMessage("");
+		defect.setToolName(PLUGIN_NAME);
+		defect.setAnalysisType(config.getAnalysisType().name());
+
+		for (DPRComment defectInfo : defectInfos) {
+			Occurence occ = new Occurence();
+			occ.setStartLine(defectInfo.getStartLine());
+			occ.setEndLine(defectInfo.getEndLine());
+			occ.setStringValue("DPR");
+			occ.setMessage(defectInfo.getReviewComment());
+
+			defect.addOccurence(occ);
+		}
+
+		return defect;
+	}
+
+	static String getFullCommentWithoutLineSeparator(String comment) {
+		return comment.replaceAll("(\r\n|\n)", "").replaceAll("\\s+", " ").trim();
```

Comment:
Don't we need to replace " " instead of ""?
eg)
abc
de
=> abcde

By the way, why do we need replace new line?
I think a user wants to see exactly what they wrote.